velox/exec/Task.h

```diff
@@ -194,7 +194,12 @@ class Task : public std::enable_shared_from_this<Task> {
   /// of buffers. No more calls are expected after the call with noMoreBuffers
   /// == true, but occasionally the caller might resend it, so calls
   /// received after a call with noMoreBuffers == true are ignored.
-  void updateBroadcastOutputBuffers(int numBuffers, bool noMoreBuffers);
+  /// @return kSuccess if update was successful.
+  ///         kNoOp if noMoreBuffers was previously set to true.
+  ///         kBuffersNotFound if buffer was not found for a given task.
+  UpdateBroadcastStatus updateBroadcastOutputBuffers(
```

Comment:
@karteekmurthys @aditi-pandit 

I took another look at this PR and the corresponding PrestoDB PR. I looks like Task::updateBroadcastOutputBuffers API doesn't need to return anything. This API can continue to be void and just be modified to not throw if task fails while this API is executing. This change won't need any additional changes in Prestissimo.

Notice how similar condition is handled in TaskManager::getDataForResultRequests(resultRequests) in presto-native-execution/presto_cpp/main/TaskManager.cpp. This call doesn't terminate execution of TaskManager::createOrUpdateTask with early return, hence, there is not need for broadcast buffer logic to do so.

Reply:
> Notice how similar condition is handled in TaskManager::getDataForResultRequests(resultRequests) in presto-native-execution/presto_cpp/main/TaskManager.cpp. This call doesn't terminate execution of TaskManager::createOrUpdateTask with early return, hence, there is not need for broadcast buffer logic to do so.

Agree, we may not return early but we do setValue in the `promise` which triggers the `future` that handled here: https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/TaskResource.cpp#L415. This promise and future in ResultRequests are created by this call `TaskManager::getResults()`. 
So we do take action if TaskManager::getDataForResultRequests(resultRequests.

Reply:
Is there a similar action we need to take in case of a failure to update broadcast buffers?

Reply:
So I am following what TaskManager::createOrUpdateTask() does if it encountered error:
https://github.com/prestodb/presto/blob/master/presto-native-execution/presto_cpp/main/TaskManager.cpp#L277

Reply:
I see. This is not necessary though. Think about what happens if the task fails right after updateBroadcastOutputBuffers returned. The processing in the TaskManager::createOrUpdateTask() will continue as if nothing happened. Hence, it doesn't matter whether task failure during updateBroadcastOutputBuffers triggers early return or not.

Reply:
Alright, I will revert changes to Task::UpdateBroadCastBuffers call. I won't take any action on broadcast failure.